language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/Value.scala

```diff
@@ -75,7 +76,9 @@ object DamlCodecs extends encoding.ValuePrimitiveEncoding[Value] {
   }
 
   implicit override val valueNumeric: Value[P.Numeric] =
-    fromArgumentValueFuns(_.numeric map BigDecimal.exact, bd => VSum.Numeric(bd.toString))
+    fromArgumentValueFuns(
+      _.numeric map lf.Numeric.assertFromString,
```

Comment:
Use `fromString`, `toOption`, and `flatMap` instead.

Reply:
Duh, somehow I didn’t realize that `_.numeric` returns an `Option`. That’s what I get for trying to avoid intellij for this small patch. Thanks for catching it! That’s much nicer.